### PR TITLE
Rap4002

### PR DESCRIPTION
**Pull request by scarabus**

### CODE REVIEW

File: Libs/CodeGen/ApiGen/src/main/resources/api/doc.api [hunk 1]
```diff
@@ -114,6 +114,10 @@ api(Doc) {
     @entitle=/data/write/$f(docUri)
     @public List(String) deleteDocsByUriPrefix(String docUri);
    
+    [Removes a folder and its contents recursively, including empty subfolders. Validates entitlement on individual docs and folders. Returns a list of the documents and folders removed.]
```
Comment (jonathan-major) [inline Libs/CodeGen/ApiGen/src/main/resources/api/doc.api]: @dukenguyen  can you review this api change please?

File: Libs/RaptureCore/src/main/java/rapture/series/children/ChildrenRepo.java [hunk 2]
```diff
@@ -75,8 +75,30 @@ public void registerChild(String parentName, String childName, boolean isFolder)
...
     public List<RaptureFolderInfo> getChildren(String dirName) {
+        // is it a file?
+        List<RaptureFolderInfo> folderInfoList = new ArrayList<>();
+
+        int lio = dirName.lastIndexOf('/');
+        if (lio > 0) {
+            String fileKey = ChildKeyUtil.createRowKey(dirName.substring(0, lio));
+            List<SeriesValue> points = getPoints(fileKey);
+            if (!points.isEmpty()) {
+                String expect = "//FILE//"+dirName.substring(lio+1);
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/series/children/ChildrenRepo.java]: should this be FILE?

Comment (scarabus) [inline Libs/RaptureCore/src/main/java/rapture/series/children/ChildrenRepo.java]: That's the expected value as stored.

File: Libs/RaptureAddins/MongoDb/src/main/java/rapture/series/mongo/MongoSeriesStore.java [hunk 3]
```diff
@@ -154,6 +154,7 @@ private void saveDocument(String key, String column, Object val) {
         try {
             @SuppressWarnings("unused")
             Document ret = collection.findOneAndUpdate(dbkey, dbval, options);
+            System.out.println(ret.toJson());
```
Comment (jonathan-major) [inline Libs/RaptureAddins/MongoDb/src/main/java/rapture/series/mongo/MongoSeriesStore.java]: this has been added in again

Comment (scarabus) [inline Libs/RaptureAddins/MongoDb/src/main/java/rapture/series/mongo/MongoSeriesStore.java]: Sigh. Yeah I have a backlog of commits.

**Pull request closed without merging**